zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -120,29 +131,179 @@ public static Either<ProblemDetail, ProcessInstanceSort> toSearchQuerySort(
         } else if ("endDate".equals(field)) {
           builder.endDate();
         } else {
-          throw new RuntimeException("unkown sortBy " + field);
+          // Collect validation error for unknown sortBy field
+          validationErrors.add(ProblemDetail.forStatusAndDetail(
+              HttpStatus.BAD_REQUEST, "Unknown sortBy field: " + field));
+          continue;
         }
 
         if ("asc".equalsIgnoreCase(order)) {
           builder.asc();
         } else if ("desc".equalsIgnoreCase(order)) {
           builder.desc();
         } else {
-          throw new RuntimeException("unkown sortOrder " + order);
+          // Collect validation error for unknown sortOrder
+          validationErrors.add(ProblemDetail.forStatusAndDetail(
+              HttpStatus.BAD_REQUEST, "Unknown sortOrder: " + order));
```

Comment:
🔧 We could add the `field` name to the error message as well so it's easier to locate the input error.